### PR TITLE
Feature/enemy refactored spawner to read config files 

### PR DESCRIPTION
**Pull request by kk4w4i**
To improve modularity and to adjust the spawner to the new config layout.

### CODE REVIEW

File: source/core/assets/levels/earth/main-area/entities.json [hunk 1]
```diff
@@ -6,28 +6,27 @@
...
     {
-      "health": 20,
-      "baseAttack": 10,
-      "speed": 5,
-      "behaviour": "PTE",
-      "spritePath": "images/enemy/base_enemy.atlas",
+      "wave1" : [
+        10,
+        0,
+        0
```
Comment (rowan-gray) [inline source/core/assets/levels/earth/main-area/entities.json:14]: Consider making each value map to a descriptive name to make it easier to modify and add to the config files.

File: source/core/assets/configs/player.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "health": 100,
+  "health": 10000000000,
```
Comment (rowan-gray) [inline source/core/assets/configs/player.json]: Revert this before merging into main please.

File: source/core/src/test/com/csse3200/game/components/npc/SpawnerComponentTest.java [hunk 3]
```diff
@@ -36,61 +37,60 @@ public void setUp() {
...
+//     */
+//    @Test
+//    public void testSpawnEnemies() {
+//        // Prepare
+//        SpawnerComponent spawner = new SpawnerComponent();
+//        int meleeCount = 5;
+//        int rangedCount = 3;
+//
+//        // Set up mock behavior
+//        when(gameTime.getTime()).thenReturn(0L);
```
Comment (rowan-gray) [inline source/core/src/test/com/csse3200/game/components/npc/SpawnerComponentTest.java:51]: Remove or fix these tests in the future please.

**Review by rowan-gray — APPROVED** [review]
Please read through my comments for feedback. There is one change I would like you to make before merging, but it's not a big enough deal to not approve since it will take 1 minute.

**Review by rowan-gray — CHANGES_REQUESTED** [review]
Nvm it crashes when going to the second planet. Please fix 😄

**Review by rowan-gray — APPROVED** [review]
All good now 😄

File: source/core/src/test/com/csse3200/game/components/npc/SpawnerComponentTest.java (hunk 3 above)
Comment (kk4w4i) [reply source/core/src/test/com/csse3200/game/components/npc/SpawnerComponentTest.java:51]: yep this will be worked on

**Pull request merged**